src/pytorch_lightning/core/mixins/device_dtype_mixin.py

```diff
@@ -189,14 +189,16 @@ def half(self) -> Self:
     def __update_properties(
         self, device: Optional[torch.device] = None, dtype: Optional[Union[str, torch.dtype]] = None
     ) -> None:
-        def apply_fn(module: Union["DeviceDtypeModuleMixin", Module]) -> None:
-            # TODO: Find why `isinstance(module, DeviceDtypeModuleMixin)` doesn't
-            # work when using `init_meta_context`.
-            if not isinstance(module, (DeviceDtypeModuleMixin, pl.LightningModule)):
-                return
+        def apply_fn(module: Union["DeviceDtypeModuleMixin", pl.LightningModule]) -> None:
             if device is not None:
                 module._device = device
             if dtype is not None:
                 module._dtype = dtype
 
-        self.apply(apply_fn)
+        # TODO: Find why `isinstance(module, DeviceDtypeModuleMixin)` doesn't
+        # work when using `init_meta_context`.
+        for m in self.modules():
+            if isinstance(m, (DeviceDtypeModuleMixin, pl.LightningModule)):
```

Comment:
```suggestion
            if isinstance(m, DeviceDtypeModuleMixin):
```
should be sufficient I guess as `LightningModule` inherits from `DeviceDtypeModuleMixin`:

https://github.com/Lightning-AI/lightning/blob/511f1a651506af2e12e15346de0a3715fed7e814/src/pytorch_lightning/core/module.py#L60